Improved credential handling

nzqo opened the pull request:
- Added environment variables for fallback of admin user and password
- Added capability to specify hash instead of cleartext passwords
- Avoid storing passwords in running beef instance; instead compare hashes

# Pull Request

Thanks for submitting a PR! Please fill in this template where appropriate:

## Category
Core Functionality

## Feature/Issue Description
**Q:** Please give a brief summary of your feature/fix
**A:** I added two fallback methods of credential specification.
    - Instead of specifying credentials in config.yaml, users may now opt to use environment variables.
    - Instead of using a password, they may also directly specify a bcrypt hash instead (works
       for both variants, config and env vars)
       
   I also modified the "default" credential check to never allow the password "beef".
   The password is now also not stored in the config instance anymore. Instead, only the hash is stored
   and used for comparison.

**Q:** Give a technical rundown of what you have changed (if applicable)
**A:** Made it so that appropriate env variables are read into config and cleared password. Introduced bcrypt gem to allow for proper hashing.

## Test Cases
**Q:** Describe your test cases, what you have covered and if there are any use cases that still need addressing.
**A:**

## Wiki Page
I believe the error message and comment in the config suffice, probably. If anything, one could update the recommendations. This change isn't breaking anyway though.

jcrew99 reviewed (commented):
Looks good so far, will need to properly step through it soon to confirm.
Do you have any thoughts on this change? @bcoles

nzqo commented on Gemfile:
Also updated to @bcoles new changes and added a version specifier.

nzqo commented:
I fixed a few operator precedence errors that broke using the hash directly as variable and added a clear error message in case someone messes this up. For reference, it should now be possible to do both

```bash
BEEF_USER=beef BEEF_PASSWD=test ./beef
```

as well as (with any bcrypt hash of the password) do something like

```bash
BEEF_USER=beef BEEF_PWHASH='$2y$12$7W3z3GwHvuq20JfTbNX.ne4.34NdqBSHLCmVaNIN8BH61YJ/ycJo.' ./beef
``` 

It might make sense to add this information to the wiki
